packages/notifications/push/amplify_push_notifications/lib/src/amplify_push_notifications_impl.dart

```diff
@@ -274,40 +259,53 @@ class AmplifyPushNotifications extends PushNotificationsPluginInterface {
 
   void _recordAnalyticsForLaunchNotification(
     PushNotificationMessage launchNotification,
-  ) {
-    _serviceProviderClient.recordNotificationEvent(
-      eventType: PinpointEventType.notificationOpened,
-      notification: launchNotification,
-    );
-  }
+  ) =>
+      _serviceProviderClient.recordNotificationEvent(
+        eventType: PinpointEventType.notificationOpened,
+        notification: launchNotification,
+      );
 }
 
 class _PushNotificationsFlutterApi implements PushNotificationsFlutterApi {
   _PushNotificationsFlutterApi() {
     PushNotificationsFlutterApi.setup(this);
   }
+  final _eventQueue = <Map<Object?, Object?>>[];
 
   final _onNotificationReceivedInBackgroundCallbacks =
       <OnRemoteMessageCallback>[];
 
-  void registerOnReceivedInBackgroundCallback(
+  Future<void> registerOnReceivedInBackgroundCallback(
     OnRemoteMessageCallback callback,
-  ) {
+  ) async {
     _onNotificationReceivedInBackgroundCallbacks.add(callback);
+    await _flushEvents();
```

Comment:
nit: Does `_flushEvents()` need to be `await`'ed for? the call of `registerOnReceivedInBackgroundCallback` at line 88 seems not need to wait for it to complete.

Reply:
right, changed to unawaited